### ISSUE TITLE
Query and confusions; Please help

### ISSUE THREAD

**Issue by shailabhr** [fanglab/nanodisco#44]
Hi,
I am still new to the Linux, GitHub, and coding thing. I read through the detailed documentation of the Nanodisco. I have already installed the singularity and nanodisco that is working. I did that with the example E. coli data. 
The Q&A section in the [documentation](https://nanodisco.readthedocs.io/en/latest/faq.html) and also your paper say that nanodisco needs Albacore 2.3.4  but then it also mentions at some places about the guppy basecaller version. I have Guppy 6.0.7 installed on my linux system. 
1. Should I use guppy 6.0.7 to basecall the data? What config file should I use for basecalling? (dna_r9.4.1_450bps_modbases_5mc_hac.cfg, dna_r9.4.1_450bps_modbases_5mc_hac_prom.cfg etc.)
2. If you suggest to use Albacore, would you please send me a link to install the proper version? I see there are several albacore codes on github. I was not sure which one I should be using. 
I have multiplexed sequenced data for which I want to find the m5C and m6A methylation (methylation typing and fine typing). 
I highly appreciate your help. 
Thank you.

**Comment by touala** [comment]
Hi @shailabhr,

Thank you for trying `nanodisco`.

You should use the latest version of Guppy available to you. `nanodisco` will likely work with any Albacore or Guppy version but the better basecalling accuracy from latest software should help on upstream analysis (e.g. _de novo_ assembly). Regarding the config file, I recommend using the one with the best accuracy possible for the flow cell/kit combination you used (e.g. dna_r9.4.1_450bps_sup.cfg). `nanodisco` is not designed to be used with *modbases*.cfg files. We also recommend using the same basecalling configuration for all datasets analyzed in a project.

Please let me know if you have other questions.

Best,

Alan

**Comment by shailabhr** [comment]
Hi Alan,
Thank you for your quick response and I am sorry for replying late. As you suggested, I generated the basecalled Fast5 files with SUP basecalling with `--fast5_out` and also performed barcode trimming. 
Step 1: preprocessing
`nanodisco preprocess -p 10 -f dataset/ctrl -s cu_ctrl -o analysis/prep_ctrl -r reference/dag20.fasta` (I used the reference genome from Ensembl bacteria)
`nanodisco preprocess -p 10 -f dataset/5cu -s 5cu -o analysis/prep_5cu -r reference/dag20.fasta`
It did produce .fasta, .bam and .bam.bai files.
Step 2: Genomic Chunk Information
`nanodisco chunk_info -r reference/dag20.fasta `
it showed the number of chunks: 747
Step: 3 Computing current differences
`nanodisco difference -nj 6 -nc 3 -p 20 -f 1 -l 747 -i analysis/prep_5cu -o analysis/diff_5cu -w 5cu -n cu_ctrl -r reference/dag20.fasta`
At this step, I did not know what numbers to be used for  `-f`  and `-l`. If my number of chunks is 747 what numbers should I use for `-f` and `-l`? How do you choose those numbers? 
At step 3, it generated more than 300 gb files and stopped at the end as I had no space left in SSD. Is it normal to produce such large files? I am going to reduce the numbers for `-nj` and `-nc` as it was getting overloaded for 64gb ram. 
Also, please see if I have used the difference command correctly. WGA refers to the sample under study and Native refers to the control sample, right? Ctrl is my control sample whereas the 5cu is the sample where I am trying to find methylation changes. I pasted all the output files from preprocessing step in prep_5cu folder as I had created separate folders initially. 
Thank you again for being so much helpful. 
Thank you. 
Sincerely,
Shailabh

**Comment by touala** [comment]
Hi Shailabh,

I don't see any red flags in the commands you run, although I'm not sure if the WGA/native options in `nanodisco difference` are correct. Usually, the "control" dataset will be from a WGA library preparation where all DNA modification are effectively stripped out, and the native dataset will be the one with the DNA modification signal still present. Does it match your experiment? Other experimental design are possible but the interpretation of results won't be straight forward. 

Regarding `nanodisco difference` optimization, using `-f` and `-l` as first and last chunk of the genome is more appropriate for large server execution with a lot of RAM. But I usually prefer to run it with smaller scope, and only processing a couple of chunks for each command (e.g. `-nj 2 -nc 5 -p 5 -f 1 -l 50` will only use 10 threads and less memory). Once all sub-commands are successfully finished you can merge all the output with `nanodisco merge`. You can start running  `-nj 1 -nc 5 -p 5 -f 1 -l 5`, monitor your system and adapt accordingly (without including chunks already processed in the next batch, e.g.  `-nj 2 -nc 5 -p 5 -f 6 -l 15`).

Regarding the storage issue, this can happened when you run many jobs in parallel as temporary files can be large. But it could also be due to high genomic coverage of your dataset. In our study, we didn't see much improvement in accuracy at 150x and above (see [Data requirement](https://nanodisco.readthedocs.io/en/latest/data_requirement.html)). If it's your case, downsampling the input `.fasta` could help. In theory, the native dataset can be randomly sampled (native should be even coverage), but if you have a WGA dataset as control then downsampling reads at coverage peaks should be better (WGA can have amplification bias).

Lastly, it should be fine to move preprocessed files around but keep in mind that the fasta files contains an index (in the read names/header lines) linking the read to the fast5 for later processing in `nanodisco difference`.

Regards,

Alan

**Comment by shailabhr** [comment]
Thank you, Alan, for responding so quickly. This was greatly helpful. 
I understood the genomic chunk thing. I will start with `-f 1 -l 5`. Even if I choose a small number here, it will process the entire 740 chunks, right? 
As you mentioned, the storage issue will also be fine if I reduce the number of jobs in parallel. 
I am still confused about the WGA and native dataset. 
Basically, I ran three experiments namely control, 5cu, and 15 cu. I prepared a library for all three and did the sequencing. The control will have some modifications naturally, however, I want to know what are new modifications happening in the other two conditions as compared to the control. If I strip out all the modifications from control, we cannot see the actual difference between 5cu and control. Similarly the difference between 15cu and control. 
OR, there should be another dataset with stripped out DNA modifications and then find the difference between this (stripped out control WGA) and all three datasets (ctrl, 5cu &15cu) separately. And then, further, find out the difference between 5cu and control. 
If we do it the latter way, how do we get the control dataset with a stripped-out modification signal?
I am fairly new to bioinformatics and I am sorry for the very basic questions. Thank you for being patient and answering my questions. I really appreciate it.
Thank you.
Sincerely,
Shailabh

**Comment by touala** [comment]
With `-nj 1 -nc 5 -f 1 -l 5`, only chunks 1 to 5 will be processed. You can either run multiple commands yourself by incrementing those values (i.e. then `-f 6 -l 10`, `-f 11 -l 15`, etc.), or specify the full range and let it handle the rest `-nj 1 -nc 5 -p 5 -f 1 -l 740` (run one job at a time with `-nj`, it will process 5 chunks and stop with `-nc`, then the next job will process another 5 chunks, etc.). All three option (`-nj`, `-nc`, `-p`) impact the RAM usage. The first solution is better for monitoring the processing.

Thank you for providing additional information regarding your experimental design. `Nanodisco` is design to detect methylation motifs present in prokaryote genomes. To be sure that we are on the same page, I'm considering that a methylation motif is a short nucleotide sequence with all their genomic occurrence modified (e.g. ~100% of CCWGG sites). I'm assuming that only the growing condition differ between dataset and that the genomes are nearly identical, right?

Do you already know what motifs are modified in your `control` dataset? If not, you should be able to find which motifs are different between two datasets (gain/loss of a motif) but you won't be able to see which one are common. Do you expect that different methylation motifs will be present depending on the experimental condition?

Regards,

Alan

**Comment by shailabhr** [comment]
Thank you again for replying. 
Yes, my genomes are identical. It is the same organism growing under different conditions. I do not know the modified motifs in my control dataset. 
Actually, for half of my DNA sample, I did the bisulfite sequencing. We have m5C methylation across cpg, chh and chg islands. I see the percentage methylation change for every gene. 
For half of my DNA sample, I did nanopore sequencing. I see nanodisco does not give out the results in terms of the same islands, however, will I know which genes have methylation changes?  Moreover, there will be m6A and m4C methylation information from Nanopore, that could be more informative. 
Currently, I thought I will try to analyze it with Nanodisco first and then will try to match it with bisulfite sequencing data. 
In my situation, which dataset I should use as WGA and which one as a native? 
I hope it is not too confusing. Thank you for helping me.

**Comment by shailabhr** [comment]
Hi,
Since I do not have the WGA dataset, I think I won't be able to use Nanodisco. I completed my experiments and sequencing before going through the data analysis tools. I am going to explore about tools that does not require WGA dataset. I would appreciate any suggestions from you. Thank you. 
Sincerely,
Shailabh

**Comment by touala** [comment]
Hi @shailabhr,

Your last message is on point. I doubt you can easily find meaningful information from `nanodisco` analysis except if the difference in the analysis are an MTase KO for example. If you want instead to simply look for methylation motifs then generating a WGA would be necessary. What did you end up doing?

I'm closing the issue but feel free to reopen or submit a new issue if necessary.

Best,

Alan
